### PR TITLE
Fix 'NoneType' object has no attribute when querying during Session not available retry

### PR DESCRIPTION
**Pull request by bambriz**
# Description

When Session Retry, Timeout Failover, and Endpoint Discovery retry policies are triggered and do no successfully retry during a Query, a bug occurs that raises a 'NoneType' object has no attribute from a request object. This was caused from the request to fetch the queried items returning back to the base request to start the query. Instead of returning the Cosmos Exception from not successfully retrying, the retry would happen again for the base query request, but the base query request does not have a reference to the Request Object (This is where the 'NoneType' object has no attribute error comes from). This is fixed by allowing the cosmos exception to surface by not retrying at the base request since we already used up the retry attempts while fetching the items during the query. 

```mermaid
    graph TD
    A[Execute Base Query Request] --> B[Fetch Items Per Page]
    B --> C[Request Per Page of Queried Items]
    C -->|Retry Fails| D[Exception]
    D --> E[Return to Base Execution]
    E -->|Check Retry| F[Nonetype Exception]


```

```mermaid
    graph TD
    A[Execute Base Query Request] --> B[Fetch Items Per Page]
    B --> C[Request Per Page of Queried Items]
    C -->|Retry Fails| D[Cosmos Exception]
    D --> E[Exception Surfaces to User]

```

### CODE REVIEW

File: sdk/cosmos/azure-cosmos/test/test_query.py [hunk 1]
```diff
@@ -12,7 +12,7 @@
 import azure.cosmos.cosmos_client as cosmos_client
 import azure.cosmos.exceptions as exceptions
 import test_config
-from azure.cosmos import http_constants, DatabaseProxy
+from azure.cosmos import http_constants, DatabaseProxy,_retry_utility, _endpoint_discovery_retry_policy
```
Comment (allenkim0129) [inline sdk/cosmos/azure-cosmos/test/test_query.py:15]: `_retry_utility` has already been imported in line 11. Please remove this and use `retry_utility` instead

File: sdk/cosmos/azure-cosmos/test/test_query_async.py [hunk 2]
```diff
@@ -12,9 +12,9 @@
 import azure.cosmos.aio._retry_utility_async as retry_utility
 import azure.cosmos.exceptions as exceptions
 import test_config
-from azure.cosmos import http_constants
+from azure.cosmos import http_constants, _endpoint_discovery_retry_policy
 from azure.cosmos._execution_context.query_execution_info import _PartitionedQueryExecutionInfo
-from azure.cosmos.aio import CosmosClient, DatabaseProxy, ContainerProxy
+from azure.cosmos.aio import CosmosClient, DatabaseProxy, ContainerProxy, _retry_utility_async
```
Comment (allenkim0129) [inline sdk/cosmos/azure-cosmos/test/test_query_async.py:17]: `_retry_utility_async` has been imported already in line 12. You can remove this import and use `retry_utility` instead

File: sdk/cosmos/azure-cosmos/test/test_query_async.py [hunk 3]
```diff
@@ -936,6 +936,91 @@ async def query_items(database):
...
+            query = "SELECT * FROM c"
+            items = created_collection.query_items(
+                query=query,
+                enable_cross_partition_query=True
+            )
+            fetch_results = [item async for item in items]
+        except exceptions.CosmosHttpResponseError as e:
+            assert e.status_code == 404
+            assert e.sub_status == 1002
+        _retry_utility_async.ExecuteFunctionAsync = self.OriginalExecuteFunction
```
Comment (allenkim0129) [inline sdk/cosmos/azure-cosmos/test/test_query_async.py:966]: Why was this assignment needed when you set `ExecuteFunctionAsync` in line 968 again?

Please remove this line if it is not necessary.

**Review by allenkim0129 — CHANGES_REQUESTED** [review]

**Review by philipthomas-MSFT — APPROVED** [review]

**Review by simorenoh — APPROVED** [review]

**Review by allenkim0129 — APPROVED** [review]

**Pull request merged**